### PR TITLE
Bug 2115899: BuildConfig form: Change image repository to image registry

### PR DESCRIPTION
**Pull request by jerolimov**
**Fixes**: 
https://bugzilla.redhat.com/show_bug.cgi?id=2115899

**Analysis / Root cause**: 
OpenShift is all about standard container images and standard container image registries, btw. called registries, not repositories.

See also https://docs.openshift.com/container-platform/4.10/registry/index.html

**Solution Description**: 
Updated the labels:

1. On "Import container image" page update the helptext below the `Image name from external registry` from `private repository` to `private registry`
    
3. On the BuildConfig ediit page:
    1. Updated `Build from` option `Docker image` to `External container image`.
        
        The other options for `Build from` are `None`, `Image Stream Tag`, and `Image Stream Image` are not changed.
    4. Updated `Push to` option `Docker image` to `External container image`.
        
        The other options for `Build from` are `None` and `Image Stream Tag` are not changed.
    3. When these options are selected there is shown the same input field below select.
        Changed this title from `Docker image repository` to `Image registry` for both!

**Screen shots / Gifs for design review**: 
| Page | Before | After |
|---|---|---|
| Add > Import from container | before-image | after-image |
| Edit BuildConfig | buildconfig-before | buildconfig-after |

**Unit test coverage report**: 
Unchanged

**Test setup:**
Unchanged

**Browser conformance**: 
- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

/cc @serenamarie125

### CODE REVIEW

File: frontend/packages/dev-console/locales/en/devconsole.json [hunk 1]
```diff
@@ -112,11 +112,11 @@
   "Enter the arguments that will be appended to the command.": "Enter the arguments that will be appended to the command.",
   "Image stream tag": "Image stream tag",
   "Image stream image": "Image stream image",
-  "Docker image repository": "Docker image repository",
+  "Image repository": "Image repository",
```
Comment (jerolimov) [inline frontend/packages/dev-console/locales/en/devconsole.json:115]: Missed this, will update to Image registry here as well:

```suggestion
  "Image repository": "Image repository",
```

**Review by serenamarie125 — APPROVED** [review]

**Comment by serenamarie125** [comment]
LGTM!  As discussed, we should be using image registry, not image repository.  Could you update the screenshot you have for the Edit BuildConfig?  Thanks!

**Comment by jerolimov** [comment]
> LGTM! As discussed, we should be using image registry, not image repository. Could you update the screenshot you have for the Edit BuildConfig? Thanks!

Updated the "image repository" to "image registry" on the BuildConfig edit page as well.

Updated the Solution Description section and screenshots in my first comment.

/cc @debsmita1 @invincibleJai

**Comment by invincibleJai** [comment]
@jerolimov looks like frontend job needs some ❤️

File: frontend/packages/dev-console/src/components/buildconfig/sections/ImagesSection.tsx [hunk 2]
```diff
@@ -89,18 +89,18 @@ const ImagesSection: React.FC<{}> = () => {
           none: t('devconsole~None'),
           imageStreamTag: t('devconsole~Image Stream Tag'),
           imageStreamImage: t('devconsole~Image Stream Image'),
-          dockerImage: t('devconsole~Docker image'),
+          dockerImage: t('devconsole~External container image'),
```
Comment (invincibleJai) [inline frontend/packages/dev-console/src/components/buildconfig/sections/ImagesSection.tsx:92]: should image be Image[upper case I] as we have above `Image Stream Image`

image

Comment (jerolimov) [reply frontend/packages/dev-console/src/components/buildconfig/sections/ImagesSection.tsx:92]: My thought is that IST and ISI are resources, while External container image isn't. Wdyt?

**Pull request merged**